agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/LoggerExporter.java

```diff
@@ -0,0 +1,359 @@
+import com.azure.monitor.opentelemetry.exporter.implementation.builders.Exceptions;
+import com.azure.monitor.opentelemetry.exporter.implementation.builders.MessageTelemetryBuilder;
+import com.azure.monitor.opentelemetry.exporter.implementation.logging.OperationLogger;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.ContextTagKeys;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.SeverityLevel;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.FormattedTime;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.TelemetryUtil;
+import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryClient;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.instrumentation.api.config.Config;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.logs.data.LogData;
+import io.opentelemetry.sdk.logs.data.Severity;
+import io.opentelemetry.sdk.logs.export.LogExporter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggerExporter implements LogExporter {
+
+  private static final Logger logger = LoggerFactory.getLogger(LoggerExporter.class);
+
+  private static final AttributeKey<String> AI_OPERATION_NAME_KEY =
+      AttributeKey.stringKey("applicationinsights.internal.operation_name");
+
+  private static final OperationLogger exportingLogLogger =
+      new OperationLogger(Exporter.class, "Exporting log");
+
+  private final TelemetryClient telemetryClient;
+  private final AtomicBoolean stopped = new AtomicBoolean();
+
+  public LoggerExporter(TelemetryClient telemetryClient) {
+    this.telemetryClient = telemetryClient;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<LogData> logs) {
+    if (CoreUtils.isNullOrEmpty(TelemetryClient.getActive().getInstrumentationKey())) {
+      logger.debug("Instrumentation key is null or empty. Fail to export logs.");
+      return CompletableResultCode.ofFailure();
+    }
+
+    if (stopped.get()) {
+      logger.debug("LoggerExporter has been stopped. Fail to export logs.");
+      return CompletableResultCode.ofFailure();
+    }
+
+    boolean failure = false;
+    for (LogData log : logs) {
+      try {
+        internalExport(log);
+        exportingLogLogger.recordSuccess();
+      } catch (Throwable t) {
+        exportingLogLogger.recordFailure(t.getMessage(), t);
+        failure = true;
+      }
+    }
+
+    return failure ? CompletableResultCode.ofFailure() : CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    return CompletableResultCode.ofSuccess();
```

Comment:
Is this TODO?

Reply:
yes, need to implement it in standalone exporter later.